Makefile

```diff
@@ -138,6 +138,18 @@ test-coverage-func coverage-func: test-coverage merge-profiles
 	@echo "\033[1;34m=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-\033[0m"
 	@go tool cover -func=coverage-all.out | egrep -v "100.0[%]"
 
+agent-mock:
```

Comment:
We generally only create one target for mocking.

Reply:
I usually do the same, but I ended up following the format already in place. There's a specific target for the `Serializer` and the `Acceptor`

Reply:
Oh ok then.